dhis-2/dhis-services/dhis-service-reporting/src/test/java/org/hisp/dhis/predictor/PredictionAnalyticsDataFetcherTest.java

```diff
@@ -0,0 +1,489 @@
+
+        gridWithAttributeOptionsA.addRow()
+            .addValue( periodB.getIsoDate() )
+            .addValue( programIndicatorB.getUid() )
+            .addValue( orgUnits.get( TEST_B ).getUid() )
+            .addValue( attributeOptionComboA.getUid() )
+            .addValue( (Double) 30.0 );
+
+        gridWithAttributeOptionsB = newGridWithAttributeOptions();
+
+        gridWithAttributeOptionsB.addRow()
+            .addValue( periodA.getIsoDate() )
+            .addValue( trackedEntityAttributeA.getUid() )
+            .addValue( orgUnits.get( TEST_C ).getUid() )
+            .addValue( attributeOptionComboA.getUid() )
+            .addValue( (Double) 40.0 );
+
+        gridWithAttributeOptionsB.addRow()
+            .addValue( periodA.getIsoDate() )
+            .addValue( trackedEntityAttributeA.getUid() )
+            .addValue( orgUnits.get( TEST_D ).getUid() )
+            .addValue( attributeOptionComboA.getUid() )
+            .addValue( (Double) 50.0 );
+
+        gridWithAttributeOptionsC = newGridWithAttributeOptions();
+
+        gridWithAttributeOptionsC.addRow()
+            .addValue( periodA.getIsoDate() )
+            .addValue( trackedEntityAttributeA.getUid() )
+            .addValue( orgUnits.get( TEST_E ).getUid() )
+            .addValue( attributeOptionComboA.getUid() )
+            .addValue( (Double) 60.0 );
+
+        gridWithoutAttributeOptionsA = newGridWithoutAttributeOptions();
+
+        gridWithoutAttributeOptionsA.addRow()
+            .addValue( periodA.getIsoDate() )
+            .addValue( programAttributeAUid )
+            .addValue( orgUnits.get( TEST_B ).getUid() )
+            .addValue( (Double) 70.0 );
+
+        gridWithoutAttributeOptionsB = newGridWithoutAttributeOptions();
+
+        gridWithoutAttributeOptionsC = newGridWithoutAttributeOptions();
+
+        gridWithoutAttributeOptionsC.addRow()
+            .addValue( periodA.getIsoDate() )
+            .addValue( programAttributeAUid )
+            .addValue( orgUnits.get( TEST_E ).getUid() )
+            .addValue( (Double) 80.0 );
+
+        fetcher = new PredictionAnalyticsDataFetcher( analyticsService );
+    }
+
+    // -------------------------------------------------------------------------
+    // Tests
+    // -------------------------------------------------------------------------
+
+    @Test
+    public void testGetAocData()
+    {
+        when( analyticsService.getAggregatedDataValues( any( DataQueryParams.class ) ) )
+            .thenAnswer( p -> getMockGrid( p ) );
+
+        fetcher.init( orgUnits, new HashSet<>( periods ), new HashSet<>( attributeOptionItems ),
+            new HashSet<>( nonAttributeOptionItems ) );
+
+        for ( int i = 0; i < TEST_SIZE; i++ )
+        {
+            MapMapMap<String, Period, DimensionalItemObject, Double> aocData = fetcher.getAocData( orgUnits.get( i ) );
+
+            MapMapMap<String, Period, DimensionalItemObject, Double> expected;
+
+            switch ( i )
```

Comment:
I feel like each size should be its own test instead of switch

Reply:
I sympathize. What I wanted to do here is to fetch every one of the 1005 orgUnits in order, because that is how the PredictionAnalyticsDataFetcher is designed to be used, and be sure that the results were where they were should be, and not elsewhere. This seemed to me to be a reasonable way to do that.